tests/scripts/run-throughput-perf.py

```diff
@@ -0,0 +1,361 @@
+##########################################################################
+
+description = 'Tool to collect throughtput performance data'
+
+parser = argparse.ArgumentParser(description=description)
+
+parser.add_argument('-arch', dest='arch', default='x64')
+parser.add_argument('-configuration', dest='build_type', default='Release')
+parser.add_argument('-run_type', dest='run_type', default='rolling')
+parser.add_argument('-os', dest='operating_system', default='Windows_NT')
+parser.add_argument('-clr_root', dest='clr_root', default=None)
+parser.add_argument('-assembly_root', dest='assembly_root', default=None)
+parser.add_argument('-benchview_path', dest='benchview_path', default=None)
+
+##########################################################################
+# Helper Functions
+##########################################################################
+
+def validate_args(args):
+    """ Validate all of the arguments parsed.
+    Args:
+        args (argparser.ArgumentParser): Args parsed by the argument parser.
+    Returns:
+        (arch, build_type, clr_root, fx_root, fx_branch, fx_commit, env_script)
+            (str, str, str, str, str, str, str)
+    Notes:
+    If the arguments are valid then return them all in a tuple. If not, raise
+    an exception stating x argument is incorrect.
+    """
+
+    arch = args.arch
+    build_type = args.build_type
+    run_type = args.run_type
+    operating_system = args.operating_system
+    clr_root = args.clr_root
+    assembly_root = args.assembly_root
+    benchview_path = args.benchview_path
+
+    def validate_arg(arg, check):
+        """ Validate an individual arg
+        Args:
+           arg (str|bool): argument to be validated
+           check (lambda: x-> bool): test that returns either True or False
+                                   : based on whether the check passes.
+
+        Returns:
+           is_valid (bool): Is the argument valid?
+        """
+
+        helper = lambda item: item is not None and check(item)
+
+        if not helper(arg):
+            raise Exception('Argument: %s is not valid.' % (arg))
+
+    valid_archs = ['x86', 'x64', 'x86jit32']
+    valid_build_types = ['Release']
+    valid_run_types = ['rolling', 'private']
+    valid_os = ['Windows_NT', 'Linux']
+
+    arch = next((a for a in valid_archs if a.lower() == arch.lower()), arch)
+    build_type = next((b for b in valid_build_types if b.lower() == build_type.lower()), build_type)
+
+    validate_arg(arch, lambda item: item in valid_archs)
+    validate_arg(build_type, lambda item: item in valid_build_types)
+    validate_arg(run_type, lambda item: item in valid_run_types)
+    validate_arg(operating_system, lambda item: item in valid_os)
+
+    if clr_root is None:
+        clr_root = nth_dirname(os.path.abspath(sys.argv[0]), 3)
+    else:
+        clr_root = os.path.normpath(clr_root)
+        validate_arg(clr_root, lambda item: os.path.isdir(clr_root))
+
+    if assembly_root is None:
```

Comment:
Same comment as above, but seems worse because this could allow us to accidentally start timing dlls that are not the same version since we are not using the nuget package version.